nhs/utils/logging.py

```diff
@@ -49,32 +51,31 @@ def config_logger():
     )
 
     logging.basicConfig(handlers=[__InterceptHandler()], level=0, force=True)
-    warnings.showwarning = lambda msg, *args, **kwargs: logger.warning(msg)
+    warnings.showwarning = lambda msg, *args, **kwargs: logger.warning(str(msg))
 
 
-def log_entry_exit(*, entry=True, exit=True, level="DEBUG"):
+def log_entry_exit(*, entry: bool = True, exit: bool = True, level: str = "DEBUG") -> Callable[[Callable[..., T]], Callable[..., T]]:
```

Comment:
btw for generic types like `T`, python 3.12.4 let you use a shorter syntax
```python
def log_entry_exit[T](*, ...):
    ...
```

So you can add `[T]` between the function name and the parentheses instead of `T = TypeVar('T')`. I'm not sure if we care much about what the function is returning, I used `Any` instead of `T` but it doesn't matter that much I think.

Oh actually while you are at it do you mind changing `level` to `level: Literal["DEBUG", "WARNING"...]`? Or make an issue since this is not really a part of your PR.

TRACE
DEBUG
INFO
SUCCESS
WARNING
ERROR
CRITICAL

Reply:
Done!